BitwardenShared/UI/Vault/Vault/VaultList/VaultListView.swift

```diff
@@ -380,15 +380,6 @@ struct VaultListView: View {
             )
         )
     }
-
-    /// Requests a review of the app.
-    private func requestReview() {
-        if #available(iOS 16.0, *) {
-            Environment(\.requestReview).wrappedValue()
-        } else {
-            SKStoreReviewController.requestReview()
-        }
-    }
```

Comment:
🤔 As stated in the previous PR, SKStoreReviewController has been [deprecated in iOS 18](https://developer.apple.com/documentation/storekit/skstorereviewcontroller). Ezimet had some problems with this so and we've also tried using a `ViewModifier` with `.apply()` + `#available` to include or not the view modifier but Ezimet said it was throwing some compiler issues.
@matt-livefront @KatherineInCode do you know any workaround for this so we can use the new environment approach for iOS 16+ and use the old approach for the older iOS version?

Reply:
Not sure if it'd work but I've thought something like:
```swift
@available(iOS 16.0, *)
struct ReviewModifier: ViewModifier {
    @Environment(\.requestReview) var requestReview
    
    let afterClosure: () -> Void
    
    func body(content: Content) -> some View {
        content.onAppear {
            requestReview()
            afterClosure()
        }
    }
}

struct OldReviewModifier: ViewModifier {
    let afterClosure: () -> Void
    func body(content: Content) -> some View {
        content.onAppear {
            SKStoreReviewController.requestReview()
            afterClosure()
        }
    }
}
```
And applying it like:
```swift
.apply { view in
    if store.state.isEligibleForAppReview {
        if #available(iOS 16.0, *) {
            view.modifier(ReviewModifier(afterClosure: {
                store.send(.appReviewPromptShown)
            }))
        } else {
            view.modifier(OldReviewModifier(afterClosure: {
                store.send(.appReviewPromptShown)
            }))
        }
    }
}
```

Reply:
And another similar way:
```swift
@available(iOS 16.0, *)
struct ReviewModifier: ViewModifier {
    @Environment(\.requestReview) var requestReview
    
    let checkEligible: Bool
    let afterClosure: () -> Void
    
    func body(content: Content) -> some View {
        content
            .task(id: checkEligible) {
                if checkEligible {
                    requestReview()
                    afterClosure()
                }
            }
    }
}

struct OldReviewModifier: ViewModifier {
    let checkEligible: Bool
    let afterClosure: () -> Void

    func body(content: Content) -> some View {
        content
            .task(id: checkEligible) {
                if checkEligible {
                    SKStoreReviewController.requestReview()
                    afterClosure()
                }
            }
    }
}
```
Applying like:
```swift
.apply { view in
    if #available(iOS 16.0, *) {
        view.modifier(ReviewModifier(checkEligible: store.state.isEligibleForAppReview, afterClosure: {
            store.send(.appReviewPromptShown)
        }))
    } else {
        view.modifier(OldReviewModifier(checkEligible: store.state.isEligibleForAppReview, afterClosure: {
            store.send(.appReviewPromptShown)
        }))
    }
}
```

Reply:
The second approach worked perfectly.
Thanks @fedemkr